misc/token.go

```diff
@@ -163,6 +164,57 @@ func GenerateAstarteJWTFromPEMKey(privateKeyPEM []byte, servicesAndClaims map[As
 	return token.String(), nil
 }
 
+// GetJWTAstarteClaims returns the set of Astarte claims for an Astarte Token.
+func GetJWTAstarteClaims(rawToken string) (AstarteClaims, error) {
+	token, err := jwt.ParseString(rawToken)
+	if err != nil {
+		return AstarteClaims{}, err
+	}
+
+	ret := AstarteClaims{}
+	err = json.Unmarshal(token.RawClaims(), &ret)
+	if err != nil {
+		return AstarteClaims{}, err
+	}
+
+	return ret, nil
+}
+
+// IsJWTAstarteClaimValidForService verifies that an Astarte Token has access to the entirety
+// of a given Astarte service (i.e. `.*::.*`).
+func IsJWTAstarteClaimValidForService(token string, service AstarteService) (bool, error) {
+	claims, err := GetJWTAstarteClaims(token)
+	if err != nil {
+		return false, err
+	}
+	switch service {
+	case AppEngine:
+		return checkAnyClaimValid(claims.AppEngineAPI), nil
+	case RealmManagement:
+		return checkAnyClaimValid(claims.RealmManagement), nil
+	case Housekeeping:
+		return checkAnyClaimValid(claims.Housekeeping), nil
+	case Pairing:
+		return checkAnyClaimValid(claims.Pairing), nil
+	case Channels:
+		return checkAnyClaimValid(claims.Channels), nil
+	case Flow:
+		return checkAnyClaimValid(claims.Flow), nil
+	default:
+		return false, fmt.Errorf("unknown Astarte service %s", service.String())
+	}
```

Comment:
Not sure if we want to err on the side of strictness right now - personally I'd go with "If there's at least a claim on the service then it's ok", which covers the case we started from (i.e. users trying to use an AppEngine-only JWT with `astartectl appengine`).
If you check for equality with the all-access token you're restricting users which could have a valid token for, e.g., a single device and disallowing them to access the device. It's better to let them and possibly let the API reply `Forbidden` since that's explicit enough.